src/sunmatrix/sparse/sunmatrix_sparse.c

```diff
@@ -153,14 +142,15 @@ SUNMatrix SUNSparseMatrix(sunindextype M, sunindextype N, sunindextype NNZ,
 SUNMatrix SUNSparseFromDenseMatrix(SUNMatrix Ad, sunrealtype droptol,
                                    int sparsetype)
 {
+  SUNFunctionBegin(Ad->sunctx);
   sunindextype i, j, nnz;
   sunindextype M, N;
   SUNMatrix As;
 
-  /* check for legal sparsetype, droptol and input matrix type */
-  if ((sparsetype != CSR_MAT) && (sparsetype != CSC_MAT)) { return NULL; }
-  if (droptol < ZERO) { return NULL; }
-  if (SUNMatGetID(Ad) != SUNMATRIX_DENSE) { return NULL; }
+  SUNAssertNull(SUNMatGetID(Ad) == SUNMATRIX_DENSE, SUN_ERR_ARG_OUTOFRANGE);
+  SUNAssertNull(sparsetype == CSC_MAT || sparsetype == CSR_MAT,
+                SUN_ERR_ARG_OUTOFRANGE);
```

Comment:
Like above, should be `SUN_ERR_ARG_INCOMPATIBLE` or `SUN_ERR_ARG_WRONGTYPE` instead?

Reply:
IMO `OUTOFRANGE` makes the more sense as it is an `int` value. Even if it was an `enum` then `OUTOFRANGE` indicates it is out of the enum range.

Reply:
Makes sense, we should check that this is consistent. I think the solvers PR uses a mix of `OUTOFRANGE` and `UNCOMPATIBLE` for this kinds of checks.